src/app/services/app-service/app.service.ts

```diff
@@ -39,6 +44,8 @@ export class AppStoryService implements IAppStoryService {
 
     const comments: Array<Observable<Comment>> = [];
 
+    if (!ids ) return forkJoin(comments);
+
     ids.forEach(id => {
       const hnComment$ = this.hnBackendService.getComment(id);
       const appComment$ = this.mapToAppComment(hnComment$);
```

Comment:
toComment$

Reply:
no need for the map function as you pointed out so it eliminates the need to have the two observables